lib/client/jsx/components/plotter/plots/bar_plot.jsx

```diff
@@ -89,39 +91,43 @@ export default class BarPlot extends React.Component{
 
 export class BarPlotBar extends React.Component{
   render() {
-    var props = this.props
+    let {
+      model_name, heights, x, scale, width, ymin, ymax, color, similar,
+      highlight_names, highlighted_name, category, mouse_handler, select 
+    } = this.props;
 
     return <g className="bar">
       {
-        props.heights.map( (name,height,i) => {
+        heights.map( (name,height,i) => {
           if (height == null) return null
 
-          if (props.select != null && props.select == i)
```

Comment:
if heights are always an array I think you are ok to remove, 
```
props.select != null
```
but you might have been checking if ``` i != null && select != null ```

Reply:
Since i comes from the map() iterator it will never be null, so checking select == i implies select != null.